### ISSUE TITLE
Provider is discarded because of one corrupted file

### ISSUE THREAD

**Issue by ratoaq2** [Diaoul/subliminal#688]
```
commit 7eb7a53fe6bcaf3e01a6b44c8366faf7c96f7f1b
Author: Antoine Bertin <diaoulael@gmail.com>
Date:   Sat Sep 3 21:54:47 2016 +0200

    Back to development version
```

LegendasTV Provider is discarded because of one corrupted file:

```
> subliminal --debug --legendastv user pass download -f -l pt-BR -p legendastv -m 97 Once.Upon.a.Time.S01E12.720p.HDTV.X264-DIMENSION.mkv
DEBUG:subliminal.cli:Collecting path Once.Upon.a.Time.S01E12.720p.HDTV.X264-DIMENSION.mkv

INFO:subliminal.providers.legendastv:Downloading archive 03ae4d4d7b3fb193ce58ca4abcb946dc
DEBUG:subliminal.providers.legendastv:Identified rar archive
ERROR:subliminal.core:Unexpected error in provider 'legendastv', discarding it
Traceback (most recent call last):
  File "/home/user/workspace/git/subliminal/subliminal/core.py", line 171, in download_subtitle
    self[subtitle.provider_name].download_subtitle(subtitle)
  File "/home/user/workspace/git/subliminal/subliminal/providers/legendastv.py", line 448, in download_subtitle
    subtitle.content = fix_line_ending(subtitle.archive.content.read(subtitle.name))
  File "/home/user/venv/subliminal/local/lib/python2.7/site-packages/rarfile.py", line 629, in read
    return f.read()
  File "/home/user/venv/subliminal/local/lib/python2.7/site-packages/rarfile.py", line 1369, in read
    self._check()
  File "/home/user/venv/subliminal/local/lib/python2.7/site-packages/rarfile.py", line 1384, in _check
    raise BadRarFile("Corrupt file - CRC check failed: " + self.inf.filename)
BadRarFile: Corrupt file - CRC check failed: Once.Upon.a.Time.S01E12.720p.HDTV.X264-DIMENSION.srt
WARNING:subliminal.core:Provider 'legendastv' is discarded
WARNING:subliminal.core:Provider 'legendastv' is discarded
WARNING:subliminal.core:Provider 'legendastv' is discarded
WARNING:subliminal.core:Provider 'legendastv' is discarded
WARNING:subliminal.core:Provider 'legendastv' is discarded
WARNING:subliminal.core:Provider 'legendastv' is discarded
WARNING:subliminal.core:Provider 'legendastv' is discarded
INFO:subliminal.core:Score 334 is below min_score (348)
DEBUG:subliminal.core:Terminating initialized providers
INFO:subliminal.core:Terminating provider legendastv
INFO:subliminal.providers.legendastv:Logging out
Downloaded 0 subtitle
DEBUG:subliminal.providers.legendastv:Logged out
```

IMO, for this case the corrupted file should be discarded and the flow should continue: search the rest of subtitles, score and download.

NOTE: Originally posted the wrong log

**Comment by Diaoul** [comment]
Which errors can be ignore and which cannot? There is no way to differentiate really.

**Comment by ratoaq2** [comment]
Just some initial thoughts: 

In download_subtitle:
 - connection errors, HTTP 4xx, HTTP 5xx: discard the provider 
 - other errors: discard the subtitle

**Comment by Diaoul** [comment]
Let's say some website changes the layout, it will bug on maybe some soup.select, or regex or anything.

**Comment by ratoaq2** [comment]
Fixed by #781
